### PR TITLE
[TRA-107] Add parentSubaccountNumber endpoint for historical-pnl

### PR DESCRIPTION
**Pull request by shrenujb**
### Changelist
Add parentSubaccountNumber endpoint for historical-pnl

### Test Plan
Added tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts [hunk 1]
```diff
@@ -86,6 +86,84 @@ class HistoricalPnlController extends Controller {
...
+            ? createdOnOrAfterHeight.toString()
+            : undefined,
+          createdOnOrAfter,
+        },
+        [QueryableField.LIMIT],
+        {
+          ...DEFAULT_POSTGRES_OPTIONS,
+          orderBy: [[QueryableField.BLOCK_HEIGHT, Ordering.DESC]],
+        },
+      ),
```
Comment (Christopher-Li) [inline indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts:131]: don't you want the to pull both the childSubaccountIds and the parentSubaccountId? You're ignoring the parent now

Comment (shrenujb) [reply indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts:131]: The childSubccountIds are inclusive. So parentSubaccountNumber 0 will have childSubaccountNumbers 0, 128, 256 , ...

File: indexer/services/comlink/__tests__/controllers/api/v4/historical-pnl-controller.test.ts [hunk 2]
```diff
@@ -207,5 +207,65 @@ describe('pnlTicks-controller#V4', () => {
         ],
       });
     });
+
+    it('Get /historical-pnl/parentSubaccountNumber', async () => {
```
Comment (Christopher-Li) [inline indexer/services/comlink/__tests__/controllers/api/v4/historical-pnl-controller.test.ts:211]: can you create a test to verify that this returns an aggregated pnl of child and parent subaccount?

Comment (shrenujb) [reply indexer/services/comlink/__tests__/controllers/api/v4/historical-pnl-controller.test.ts:211]: That is exactly what this test does

File: indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts (hunk 1 above)
Comment (shrenujb) [reply indexer/services/comlink/src/controllers/api/v4/historical-pnl-controller.ts:131]: It does? https://github.com/dydxprotocol/v4-chain/blob/0dbaf0b0df4a684d58233acc275a79e6988688ff/indexer/services/comlink/src/lib/helpers.ts#L504
Is there anything more you'd like to have specified?

**Review by vincentwschau — APPROVED** [review]

**Comment by vincentwschau** [comment]
@Mergifyio backport release/indexer/v5.x

**Pull request merged**